sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -241,7 +243,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   private def listTestCases(): Seq[TestCase] = {
     listFilesRecursively(new File(inputFilePath)).map { file =>
       val resultFile = file.getAbsolutePath.replace(inputFilePath, goldenFilePath) + ".out"
-      TestCase(file.getName, file.getAbsolutePath, resultFile)
+      val absPath = file.getAbsolutePath
+      TestCase(absPath.stripPrefix(inputFilePath + File.separator), absPath, resultFile)
```

Comment:
Does this get into problems where inputFilePath might already have a trailing separator and then isn't a prefix? maybe it can't happen, but maybe it's easier to strip leading separators from the final result, directly, if that's what you mean to do.